lib/Expression/NodePrinter/DisplayAsPrinter.php

```diff
@@ -39,18 +41,32 @@ public function print(Printer $printer, Node $node, array $params): ?string
             return null;
         }
 
-        $unit = $node->as();
+        $unitNode = $node->as();
```

Comment:
This entire class is getting out of control.

Extract a dedicated unit converter and formatter instead of integrating with the shitty TimeUnit and MemoryUnit.